### PR TITLE
useSingleRequestParameter should mark parameter optional if all properties are optional

### LINKED ISSUE #11112: [REQ] [typescript-fetch] useSingleRequestParameter should mark parameter optional if all properties are optional
### Is your feature request related to a problem? Please describe.

when useSingleRequestParameter has all optional parameters, an empty object still has to be passed producing `myApiCall({})` 

## Describe the solution you'd like

Functions with all optional parameters should include a default empty object parameter to make it optional instead of `myApiCall({})` 

When the default is in place, we can instead use `myApiCall()` 

See #6477 and related #6440

## Describe alternatives you've considered

Instead of making the parameter object itself optional, we can use the solution employed in the referenced PR to default which will effectively make it optional too

### PR DESCRIPTION
**Pull request by davidgamero**
This PR uses a default value to requestParameters (= {}) if all the parameters are optional.

Fixes #11112

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov

### CODE REVIEW

**Review by amakhrov — APPROVED** [review]
Thanks for your PR. Looks great (and I even feel other TS generators would benefit from a similar change, too).

**Review by OliverMKing — APPROVED** [review]

**Comment by davidgamero** [comment]
Thank you! Just need a maintainer approval to run workflow next

**Review by macjohnny — APPROVED** [review]

**Pull request merged**
